core/domain/html_cleaner.py

```diff
@@ -162,20 +170,20 @@ def get_rte_components(html_string: str) -> List[Dict[str, Any]]:
         - id: str. The name of the component, i.e. 'oppia-noninteractive-link'.
         - customization_args: dict. Customization arg specs for the component.
     """
-    components: List[Dict[str, Any]] = []
+    components: List[ComponentsDict] = []
     soup = bs4.BeautifulSoup(html_string, 'html.parser')
     oppia_custom_tag_attrs = (
         rte_component_registry.Registry.get_tag_list_with_attrs())  # type: ignore[no-untyped-call]
     for tag_name, tag_attrs in oppia_custom_tag_attrs.items():
         component_tags = soup.find_all(name=tag_name)
         for component_tag in component_tags:
             component = {'id': tag_name}
-            customization_args = {}
+            customization_args: Dict[str, Any] = {}
             for attr in tag_attrs:
                 # Unescape special HTML characters such as '&quot;'.
                 attr_val = html.unescape(component_tag[attr])
                 customization_args[attr] = json.loads(attr_val)
 
             component['customization_args'] = customization_args
-            components.append(component)
+            components.append(component)   # type: ignore[arg-type]
```

Comment:
Previously, I tried `TypedDict` but I was not sure how to get rid of this error and used `Dict[str, Any]`. Maybe I was wrong. Here, It seems like the reason is that Mypy is assuming the type of `components` and we can't type annotate the variable with `ComponentsDict` ( on line 181 ) because that will give a `missing key` error which is quite relevant. So, what you would suggest removing this error msg.

Reply:
Rather than annotating line 181, couldn't we assert the type after the `customization_args` is formed?

Reply:
Ahh, but we can't use `TypedDict` in `isinstance` like this `assert instance(customization_args, ComponentsDict)`. Can we use `total` here in `TypedDict`? Something like this [commit](https://github.com/oppia/oppia/pull/15435/commits/1388086f88135fcdb677470734ee124eb6cbcec3) to get rid of the error.